src/services/urlAssembly.spec.ts

```diff
@@ -41,6 +42,20 @@ describe('path params', () => {
     expect(url).toBe('/simple/')
   })
 
+  test('given route with default string param NOT provided, returns route Path with string with default value interpolated', () => {
+    const [route] = createRoutes([
+      {
+        name: 'simple',
+        path: path('/simple/[?simple]', { simple: withDefault(String, 'abc') }),
+        component,
+      },
+    ])
+
+    const url = assembleUrl(route)
+
+    expect(url).toBe('/simple/abc')
+  })
```

Comment:
Ah yeah I think this answers my previous question about the setter logic. I don't think this should put the default value in the url. The default value should come from the getter recognizing the value is `undefined`.